### PR TITLE
Removes iGEMM solvers not typically picked

### LINKED ISSUE #229: Remove iGEMM solvers
The following solvers need to be removed from the library:

* ConvHipImplicitGemmV4_1x1
* ConvHipImplicitGemmV4Fwd

@asroy Please update the list if you believe some others can be removed as well

### PR DESCRIPTION
**Pull request by JehandadKhan**
Resolves issue #229 

The following solvers have been removed:

### fwd, bwd-data

- [X] ConvHipImplicitGemmV4R4Xdlops_1x1
- [X] ConvHipImplicitGemmV4R4FwdXdlops
- [X] ConvHipImplicitGemmV4_1x1
- [X] ConvHipImplicitGemmV4Fwd

### bwd-wrw solver

- [X] ConvHipImplicitGemmV4WrW
- [X] ConvHipImplicitGemmV4R4WrWXdlops

### CODE REVIEW

**Comment by atamazov** [comment]
I think now we have some dead code in RunAndMeasureSolutionBase and GetSolutionBase. Maybe some invokers, too.

**Comment by asroy** [comment]
fwd, bwd-data 
```
ConvHipImplicitGemmV4R4Xdlops_1x1
ConvHipImplicitGemmV4R4FwdXdlops
ConvHipImplicitGemmV4_1x1
ConvHipImplicitGemmV4Fwd
```
bwd-wrw solver
```
ConvHipImplicitGemmV4WrW
ConvHipImplicitGemmV4R4WrWXdlops
```
Can be removed.


@zjing14 @TejashShah please confirm

**Comment by zjing14** [comment]
Looks good.

**Comment by TejashShah** [comment]
Looks good.

File: src/solver.cpp [hunk 1]
```diff
@@ -235,10 +235,8 @@ inline SolverRegistrar::SolverRegistrar(IdRegistryData& registry)
     RegisterWithSolver(registry, ++id, ConvOclBwdWrW1x1{}, miopenConvolutionAlgoDirect);
     RegisterWithSolver(
         registry, ++id, ConvHipImplicitGemmV4R1Fwd{}, miopenConvolutionAlgoImplicitGEMM);
-    RegisterWithSolver(
-        registry, ++id, ConvHipImplicitGemmV4Fwd{}, miopenConvolutionAlgoImplicitGEMM);
-    RegisterWithSolver(
-        registry, ++id, ConvHipImplicitGemmV4_1x1{}, miopenConvolutionAlgoImplicitGEMM);
+    ++id; // removed solver ConvHipImplicitGemmV4Fwd
+    ++id; // removed solver ConvHipImplicitGemmV4_1x1
```
Comment (atamazov) [inline src/solver.cpp:239]: :+1:

**Comment by JehandadKhan** [comment]
@atamazov @asroy @zjing14 Please review again I have removed all the indicated solvers.

**Comment by JehandadKhan** [comment]
> I think now we have some dead code in RunAndMeasureSolutionBase and GetSolutionBase. Maybe some invokers, too.

@atamazov  I believe I have fixed that, please let me know if I still need to remove something.

File: src/solver/conv_hip_implicit_gemm_v4.cpp [hunk 2]
```diff
@@ -1,478 +0,0 @@
...
-    construction_parameters.g_wk.push_back(gbl_wk1);
-    construction_parameters.g_wk.push_back(gbl_wk2);
-
-    if(kernel == ImplicitGemmV4)
-    {
-        construction_parameters.kernel_file =
-            "gridwise_convolution_implicit_gemm_v4_nchw_kcyx_nkhw_lds_double_buffer.cpp";
-
-        construction_parameters.kernel_name =
-            "gridwise_convolution_implicit_gemm_v4_nchw_kcyx_nkhw_lds_double_buffer";
```
Comment (atamazov) [inline src/solver/conv_hip_implicit_gemm_v4.cpp:249]: Please find names of removed kernels in `MakeImplGemmDataInvokerFactory()` and remove them there.

**Comment by asroy** [comment]
> Unfortunately, there is some dead code in invokers. And some unused wrappers, I guess.

@JehandadKhan @atamazov I can remove these dead code after #243 is merged

**Comment by atamazov** [comment]
@asroy #243 is merged. Are you going to update this branch/PR directly?

@JehandadKhan Could you please resolve merge conflicts, thanks.

File: src/solver/conv_hip_implicit_gemm_v4.cpp (hunk 2 above)
Comment (JehandadKhan) [reply src/solver/conv_hip_implicit_gemm_v4.cpp:249]: @atamazov The kernel wrappers mentioned in this comment have been deleted.

File: src/conv/invokers/impl_gemm.cpp [hunk 3]
```diff
@@ -70,30 +70,6 @@ InvokerFactory MakeImplGemmDataInvokerFactory(const ConvolutionContext& ctx)
                         elapsed += handle.GetKernelTime();
                 }
                 // clang-format off
-                else if((kernel.GetName() == "gridwise_convolution_implicit_gemm_v4_nchw_kc1x1_nkhw_lds_double_buffer") ||
```
Comment (JehandadKhan) [inline src/conv/invokers/impl_gemm.cpp:73]: [anchor]

File: src/solver/conv_hip_implicit_gemm_v4.cpp (hunk 2 above)
Comment (JehandadKhan) [reply src/solver/conv_hip_implicit_gemm_v4.cpp:249]: The invoker logic has been [removed as well](https://github.com/ROCmSoftwarePlatform/MIOpen/pull/274/files#r447191052).

**Comment by JehandadKhan** [comment]
@atamazov , @asroy the kernel wrappers and the remaining invoker logic has been removed, please review.

**Comment by JehandadKhan** [comment]
@daniellowell  CI passes

**Comment by atamazov** [comment]
@JehandadKhan Thank you!

**Pull request merged**